<launch>
    <arg name="yaw_std_dev" default="0.001"/>
    <arg name="xyz_std_dev" default="[0.001, 0.001, 0.001]"/>
    <arg name="parent_frame" default="odom"/>
    <arg name="child_frame" default="base_link_noisy"/>
    <node pkg="hydra_ros" type="noisy_tf_publisher" name="noisy_tf_publisher_node" output="screen">
        <rosparam subst_value="true">
            yaw_std_dev: $(arg yaw_std_dev)
            xyz_std_dev: $(arg xyz_std_dev)
            noisy_tf_parent: $(arg parent_frame)
            noisy_tf_child: $(arg child_frame)
        </rosparam>
    </node>

    <arg name="robot_id" default="0"/>
    <arg name="start_gt_lcd" default="false"/>
    <arg name="gt_parent_frame" default="map"/>
    <arg name="gt_child_frame" default="base_link_gt"/>
    <node pkg="ouroboros_ros" type="vlc_server_node.py" name="vlc_server_node" required="true" output="screen">
        <param name="show_plots" value="false"/>
        <param name="lc_send_delay_s" value="15"/>
        <param name="vlc_frame_period_s" value="0.5"/>
        <param name="robot_id" value="$(arg robot_id)"/>
        <param name="config_path" value="$(find ouroboros_ros)/config/gt_vlc_server_config.yaml"/>
        <param name="fixed_frame" value="$(arg gt_parent_frame)"/>
        <param name="hint_body_frame" value="$(arg gt_child_frame)"/>
    </node>
</launch>
